pyvista/core/filters/data_set.py

```diff
@@ -1990,11 +1977,16 @@ def glyph(
             alg.SetScaleModeToDataScalingOff()
 
         if isinstance(orient, str):
-            dataset.active_vectors_name = orient
+            if scale and dataset.active_scalars_info.association == FieldAssociation.CELL:
+                prefer = 'cell'
+            else:
+                prefer = 'point'
```

Comment:
If we don't change the default preference above, we should reorder the preference here.

Reply:
I've changed the `scale` preference to be `cell`, as it was before. I don't believe anything needs to be changed here, because this merely attempts to use the same preference for the `orient` as the `scale`. If it ends up choosing a vector field which has a different FieldAssociation, an `ValueError` will be raised downstream.

Reply:
The default for `orient` will be `point` if `scale is False`. yet the default for `scale` is `cell`.  This is all a mess, but it isn't this PR's fault.  I'm okay with leaving it as you see best.

Reply:
Your PR state currently matches this behavior somewhat:

>FWIW, dataset.set_active_scalars() prefers cell by default, whereas dataset.set_active_vectors() prefers point.

So I think this is good in that light.